config_test.go

```diff
@@ -0,0 +1,243 @@
+package config
+
+import (
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/wawandco/fako"
+)
+
+const (
+	testTomlName        = "test.toml"
+	testTomlDefaultName = "test-default.toml"
+	testTomlFileDir     = "/"
+)
+
+func newFs(t *testing.T, file string) func() {
+	fs = afero.NewMemMapFs()
+	b, err := ioutil.ReadFile(file)
+	require.NoError(t, err)
+	filePath := path.Join(testTomlFileDir, configFileName)
+	require.NoError(t, afero.WriteFile(fs, filePath, b, 0644))
+	f := path.Join(os.TempDir(), filePath+".lock")
+	lockFilePath = func(p string) string {
+		return f
+	}
+	return func() {
+		os.Remove(f)
+	}
+}
+
+func printConfigFile(dir string) {
+	filePath := path.Join(dir, configFileName)
+	b, err := afero.ReadFile(fs, filePath)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+	log.Println(string(b))
+}
+
+func TestReadingConfigInDir(t *testing.T) {
+	defer newFs(t, testTomlName)()
+	conf, err := New(testTomlFileDir)
+	require.NoError(t, err)
+
+	device := Device{}
+	deviceChanges := Device{}
```

Comment:
`var deviceChanges Device` is more idomatic. Same for the other places in this file.